api/endpoints/activities.py

```diff
@@ -1,11 +1,56 @@
 """Module for Activities in Andela."""
-from flask import jsonify
+from flask import jsonify, request
 from flask_restplus import Resource
 
-from ..auth import token_required
-from ..models import Activity
+from api.auth import token_required
+from api.models import Activity
 
 
 class ActivitiesAPI(Resource):
-    """Activity Resource."""
-    pass
+    """Activity Resource to contain
+    CRUD endpoints for activities"""
+
+    @token_required
+    def post(self):
+        """ Create an activity """
+
+        payload = request.get_json()
+
+        if not payload:
+            response = jsonify({
+                "status": "fail",
+                "message": "name and description required"})
+            response.status_code = 400
+        elif "name" not in payload.keys():
+            response = jsonify({
+                    "status": "fail",
+                    "message": "Name is required to create an activity."
+                })
+            response.status_code = 400
+
+        else:
+            name = payload["name"]
+            description = payload["description"]
+
+            existing_activity = Activity.query.filter(
+                                Activity.name.ilike(name)).first()
+
+            if existing_activity:
+                response = jsonify({
+                    "message": "Activity already exists!"
+                })
+                response.status_code = 405
+
+            else:
+                activity = Activity(
```

Comment:
An activity requires an activityType.

Reply:
An activity requires a user and a society.

Reply:
@georgreen When the Activity is being created how will it be linked with the user and society as such?

Reply:
@Andretalik +1
@georgreen The models failed to establish those relationships and also I've noticed ActivityType shouldn't have the `activity_id` foreign key. It's supposed to be in the Activity model as `activity_type_id`.

Reply:
@Andretalik 
My bad on the society part come to think of it an activity won't need that but it requires a user who created it.

The models are not complete, as we code we need to adjust and add relationships or fields as required.

Reply:
@georgreen I have included the activity type. About the user have we decided on having an `added_by` field in the Activity model?